dts/arm/realtek/rtl8752h.dtsi

```diff
@@ -0,0 +1,83 @@
+#include <mem.h>
+#include <arm/armv6-m.dtsi>
+#include <freq.h>
+// #include <dt-bindings/pinctrl/rtl87x2g-pinctrl.h>
+// #include <dt-bindings/clock/rtl87x2g-clocks.h>
+// #include <zephyr/dt-bindings/gpio/gpio.h>
+// #include <zephyr/dt-bindings/adc/adc.h>
+// #include <zephyr/dt-bindings/pwm/pwm.h>
+// #include <zephyr/dt-bindings/i2c/i2c.h>
+
+/ {
+	chosen {
+		zephyr,flash = &flash;
+		zephyr,flash-controller = &fmc;
+		zephyr,sram = &app_ram_data;
+	};
+
+	cpus {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		cpu: cpu@0 {
+			device_type = "cpu";
+			compatible = "arm,cortex-m0+";
+			reg = <0>;
+			#address-cells = <1>;
+			#size-cells = <1>;
+		};
+	};
+
+	soc {
+		fmc: flash-controller@40080000 {
+			compatible = "realtek,rtl8752h-flash-controller";
+			reg = <0x40080000 0x12C>;
+			#address-cells = <1>;
+			#size-cells = <1>;
+			status = "okay";
+
+			flash: flash@800000 {
+				compatible = "soc-nv-flash";
+				write-block-size = <1>;
+				erase-block-size = <4096>;
+				reg = <0x800000 DT_SIZE_K(512)>;
```

Comment:
Move this "reg = <0x800000 DT_SIZE_K(512)>;" to board level dts, since flash-size is   differed from board, not from soc_series.

Reply:
Agree, i will move this property.